reactioncommerce/reaction
# Fix "Shipping methods not found" bug

Resolves #2265 .

There was a bug/misleading error that used to show up now and then whenever a user tried to checkout. This bug was due to a combination of:

- not showing a status message while shipping methods were being loaded.
- weak internet connections.
- a race-condition-like situation.

What this PR does is to show a loading icon + message while shipping methods are being retrieved from various APIs (e.g Shippo's). And if that retrieval fails, then this PR aims to show an appropriate error message too.

## Test instructions
1. Enable Shippo as your only shipping methods provider.
1. Open a new incognito browser window.
1. *As an anonymous/unregistered user*, add items to your cart.
1. Proceed to checkout.
1. Fill in your address and save it. You should now see a "Loading shipping methods... Please wait..." message in the shipping section.
1. Wait for a few seconds, and you'll see that the loading message in the preceding step is replaced by a list of shipping methods.
